common/cache/dialer.go

```diff
@@ -39,7 +39,7 @@ func NewDialer(options Options) (DialerFunc, error) {
 				finalIps = append(finalIps, ip)
 			}
 		}
-		if err != nil || len(finalIps) <= 0 {
+		if err != nil || len(finalIps) < 0 {
```

Comment:
Could you change finalIps to `<=0`since it can never be less than 0. Here we are checking if there are no IP addresses returned from the resolver.

Reply:
Hi @Ice3man543,

I updated the PR to check if length of finalIps is exactly 0. It doesn't make sense to check if it's less than 0. When building my PR I mistyped less than 0 instead of equals to 0. Sorry about that.

Let me know if any other change is needed from my side.